In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

pd.set_option("display.max_columns", None)

In [6]:
df = pd.read_csv("Nassau Candy Distributor.csv")

df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Country/Region,City,State/Province,Postal Code,Division,Region,Product ID,Product Name,Sales,Units,Gross Profit,Cost
0,1,US-2021-103800-CHO-MIL-31000,03-01-2024,30-06-2026,Standard Class,103800,United States,Houston,Texas,77095,Chocolate,Interior,CHO-MIL-31000,Wonka Bar - Milk Chocolate,6.50,2,4.22,2.28
1,2,US-2021-112326-CHO-TRI-54000,04-01-2024,01-07-2026,Standard Class,112326,United States,Naperville,Illinois,60540,Chocolate,Interior,CHO-TRI-54000,Wonka Bar - Triple Dazzle Caramel,7.50,2,4.90,2.60
2,3,US-2021-112326-CHO-NUT-13000,04-01-2024,01-07-2026,Standard Class,112326,United States,Naperville,Illinois,60540,Chocolate,Interior,CHO-NUT-13000,Wonka Bar - Nutty Crunch Surprise,10.47,3,7.47,3.00
3,4,US-2021-112326-CHO-SCR-58000,04-01-2024,01-07-2026,Standard Class,112326,United States,Naperville,Illinois,60540,Chocolate,Interior,CHO-SCR-58000,Wonka Bar -Scrumdiddlyumptious,10.80,3,7.50,3.30
4,5,US-2021-141817-CHO-TRI-54000,05-01-2024,05-07-2026,Standard Class,141817,United States,Philadelphia,Pennsylvania,19143,Chocolate,Atlantic,CHO-TRI-54000,Wonka Bar - Triple Dazzle Caramel,11.25,3,7.35,3.90


In [7]:
df.shape

(10194, 18)

In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 10194 entries, 0 to 10193
Data columns (total 18 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Row ID          10194 non-null  int64  
 1   Order ID        10194 non-null  str    
 2   Order Date      10194 non-null  str    
 3   Ship Date       10194 non-null  str    
 4   Ship Mode       10194 non-null  str    
 5   Customer ID     10194 non-null  int64  
 6   Country/Region  10194 non-null  str    
 7   City            10194 non-null  str    
 8   State/Province  10194 non-null  str    
 9   Postal Code     10194 non-null  str    
 10  Division        10194 non-null  str    
 11  Region          10194 non-null  str    
 12  Product ID      10194 non-null  str    
 13  Product Name    10194 non-null  str    
 14  Sales           10194 non-null  float64
 15  Units           10194 non-null  int64  
 16  Gross Profit    10194 non-null  float64
 17  Cost            10194 non-null  float64
dt

In [9]:
df.describe()

,Row ID,Customer ID,Sales,Units,Gross Profit,Cost
count,10194.000000,10194.000000,10194.000000,10194.000000,10194.000000,10194.000000
mean,5097.500000,134468.961154,13.908537,3.791838,9.166451,4.742087
std,2942.898656,20231.483007,11.341020,2.228317,6.643740,5.061647
min,1.000000,100006.000000,1.250000,1.000000,0.250000,0.600000
25%,2549.250000,117212.000000,7.200000,2.000000,4.900000,2.400000
50%,5097.500000,133550.000000,10.800000,3.000000,7.470000,3.600000
75%,7645.750000,152051.000000,18.000000,5.000000,12.250000,5.700000
max,10194.000000,192314.000000,260.000000,14.000000,130.000000,130.000000


In [10]:
df.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Country/Region', 'City', 'State/Province',
       'Postal Code', 'Division', 'Region', 'Product ID', 'Product Name',
       'Sales', 'Units', 'Gross Profit', 'Cost'],
      dtype='str')

In [11]:
df.isnull().sum()

Row ID            0
Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Country/Region    0
City              0
State/Province    0
Postal Code       0
Division          0
Region            0
Product ID        0
Product Name      0
Sales             0
Units             0
Gross Profit      0
Cost              0
dtype: int64

In [13]:
df.duplicated().sum()

np.int64(0)

In [14]:
df = df.drop_duplicates()

In [4]:
import os

print(os.getcwd())

c:\Users\klarikach\OneDrive\Documents\Nassau_Candy_Project\notebooks


In [5]:
import os

print(os.listdir())

['1_Data_cleaning.ipynb']


Checking Negative Sales


In [15]:
df[df["Sales"] < 0]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Country/Region,City,State/Province,Postal Code,Division,Region,Product ID,Product Name,Sales,Units,Gross Profit,Cost


Checking Negative Profits

In [21]:
df[df["Gross Profit"] < 0]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Country/Region,City,State/Province,Postal Code,Division,Region,Product ID,Product Name,Sales,Units,Gross Profit,Cost


Negative Cost

In [16]:
df[df["Cost"] < 0]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Country/Region,City,State/Province,Postal Code,Division,Region,Product ID,Product Name,Sales,Units,Gross Profit,Cost


Checking zero Sales

In [17]:
df[df["Sales"] == 0]

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Country/Region,City,State/Province,Postal Code,Division,Region,Product ID,Product Name,Sales,Units,Gross Profit,Cost


In [20]:
df["Order Date"] = pd.to_datetime(
    df["Order Date"],
    dayfirst=True
)

df["Ship Date"] = pd.to_datetime(
    df["Ship Date"],
    dayfirst=True
)

In [22]:
df["Year"] = df["Order Date"].dt.year

df["Month"] = df["Order Date"].dt.month_name()

df["Quarter"] = df["Order Date"].dt.quarter

KPI Creation


In [23]:
df["Gross Margin %"] = (
    df["Gross Profit"] / df["Sales"]
) * 100

In [24]:
df["Profit Per Unit"] = (
    df["Gross Profit"] / df["Units"]
)

In [25]:
df["Sales Per Unit"] = (
    df["Sales"] / df["Units"]
)

In [26]:
df["Cost Per Unit"] = (
    df["Cost"] / df["Units"]
)

In [27]:
total_sales = df["Sales"].sum()

df["Revenue Contribution %"] = (
    df["Sales"] / total_sales
) * 100

In [29]:
total_profit = df["Gross Profit"].sum()

df["Profit Contribution %"] = (
    df["Gross Profit"] / total_profit
) * 100

In [30]:
df.to_csv("cleaned_nassau_candy_distributor.csv", index=False)

In [31]:
summary = pd.DataFrame({
    "Metric": [
        "Total Records",
        "Total Products",
        "Total Divisions",
        "Total Sales",
        "Total Profit",
        "Average Margin (%)"
    ],
    "Value": [
        len(df),
        df["Product Name"].nunique(),
        df["Division"].nunique(),
        round(df["Sales"].sum(), 2),
        round(df["Gross Profit"].sum(), 2),
        round((df["Gross Profit"].sum()/df["Sales"].sum())*100, 2)
    ]
})

summary

,Metric,Value
0,Total Records,10194.00
1,Total Products,15.00
2,Total Divisions,3.00
3,Total Sales,141783.63
4,Total Profit,93442.80
5,Average Margin (%),65.91


This Nassau Candy Distributor dataset contains 10,194 transaction records across 15 unique products and 3 product divisions like Chocolates, Sugar and Other

Identified: 
Total Records: 10,194,

Total Products: 15,

Total Divisions: 3,

Total Sales: $141,783.63,

Total Gross Profit: $93,442.80,

Average Gross Margin: 65.91%.

The high average gross margin indicates that the organization generates substantial profit relative to sales revenue. However, aggregate performance alone does not reveal whether profitability is concentrated among a small number of products or divisions. Further analysis is required to identify high-performing products, margin-risk products, and divisions that contribute disproportionately to overall profitability.